src/core/algorithms/nar/des/encoded_nar.cpp

```diff
@@ -0,0 +1,102 @@
+#include "encoded_nar.h"
+
+#include "algorithms/nar/value_range.h"
+#include "model/types/types.h"
+
+namespace algos::des {
+
+using model::NAR;
+
+size_t EncodedNAR::VectorSize() {
+    return encoded_value_ranges_.size() * EncodedValueRange().kFieldCount + 1;
+}
+
+size_t EncodedNAR::FeatureCount() {
+    return encoded_value_ranges_.size();
+}
+
+// TODO: remove code duplication here
+double& EncodedNAR::operator[](size_t index) {
+    qualities_consistent_ = false;
+    if (index == 0) {
+        return implication_sign_pos_;
+    } else {
+        index--;
+        size_t feature = index / EncodedValueRange().kFieldCount;
+        size_t feature_field = index % EncodedValueRange().kFieldCount;
+        return encoded_value_ranges_[feature][feature_field];
+    }
```

Comment:
no need to have a branch here since we `return` in if. Just
```cpp
    if (index == 0) {
        return implication_sign_pos_;
    }
    index--;
    size_t feature = index / EncodedValueRange::kFieldCount;
    size_t feature_field = index % EncodedValueRange::kFieldCount;
    return encoded_value_ranges_[feature][feature_field];
```

Reply:
yea, use `::` to access static member variables